### PR TITLE
test: Add interop JWE tests with Go Jose

### LINKED ISSUE #1726: Add JWE tests interoperable with Go Jose
as a follow up to #1638

need more tests to verify encrypting JWE with Go Jose can be decrypted using Aries-framework-go's new JWE impelementation

### PR DESCRIPTION
**Pull request by Baha-sk**
closes #1726

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

**Comment by DRK3** [comment]
The semantic commit prefix "test:" might be a better fit

File: pkg/doc/jose/encrypter_decrypter_test.go [hunk 1]
```diff
@@ -159,6 +162,63 @@ func TestJWEEncryptRoundTrip(t *testing.T) {
...
+
+func convertToGoJoseRecipients(t *testing.T, keys []ecdhessubtle.ECPublicKey) []jose.Recipient {
+	t.Helper()
+
+	var joseRecipients []jose.Recipient
+
+	for _, key := range keys {
+		c := subtle.GetCurve(key.Curve)
+		gjKey := jose.Recipient{
+			Algorithm: jose.ECDH_ES_A256KW, // for now, this is the only key wrapping algorithm supported
```
Comment (DRK3) [inline pkg/doc/jose/encrypter_decrypter_test.go:208]: Not sure if you need this comment... consider one day if we add another key wrapping algorithm... unless you remember to update this comment it'll stay there and then be misleading

Comment (Baha-sk) [reply pkg/doc/jose/encrypter_decrypter_test.go:208]: I'll remove it..

Comment (Baha-sk) [reply pkg/doc/jose/encrypter_decrypter_test.go]: removed

File: pkg/doc/jose/encrypter_decrypter_test.go [hunk 2]
```diff
@@ -159,6 +162,63 @@ func TestJWEEncryptRoundTrip(t *testing.T) {
 	}
 }
 
+func TestRoundTripInteropWithGoJose(t *testing.T) {
```
Comment (troyronda) [inline pkg/doc/jose/encrypter_decrypter_test.go:165]: what about Encrypt with our package and decrypt with go-jose?

Comment (Baha-sk) [reply pkg/doc/jose/encrypter_decrypter_test.go:165]: I'll add a test

**Pull request merged**